File: ios/appletvserver/appletvserver/M3u8Download.m
//
//  M3u8Download.m
//  appletvserver
//
//  Created by zhang wei on 13-2-27.
//
//

#import "M3u8Download.h"
#import "AtvUtil.h"
#import "AppDelegate.h"
@implementation M3u8Download

@synthesize m3u8DownloadLines = _m3u8DownloadLines;
@synthesize m3u8Url = _m3u8Url;
@synthesize tempSize = _tempSize;


-(long) getDurationTempSize{
    long temp = [AtvUtil getFolderSize:AppDele.localM3u8PathPrefix];
    long result = temp-_tempSize;
    _tempSize = temp;
    return result;
}
-(void) seekDownloadLine:(NSString*) localPath{
    @synchronized(self) {
        for(int i=0;i<[_m3u8DownloadLines count];i++){
            M3u8DownloadLine* downloadLine=[_m3u8DownloadLines objectAtIndex:i];
            if([downloadLine.localPath isEqualToString:localPath]){
                _downloadIndex = i;
                break;
            }
        }
    }
}
- (M3u8DownloadLine*) getNextDownloadLine{
     @synchronized(self) {
         if(_downloadIndex<[_m3u8DownloadLines count]){
             M3u8DownloadLine* next=[_m3u8DownloadLines objectAtIndex:_downloadIndex];
             _downloadIndex++;
             return next;
         }else{
             return nil;
         }
     }
}


- (void) dealloc {
    TT_RELEASE_SAFELY(_m3u8DownloadLines);
    TT_RELEASE_SAFELY(_m3u8Url);
    [super dealloc];
}

@end

File: tags/ios/2013-04-24-1.0.1/appletv/appletv/src/main/java/com/clican/appletv/core/service/proxy/model/M3u8Download.java
package com.clican.appletv.core.service.proxy.model;

import java.util.List;

public class M3u8Download {

	private String m3u8Url;

	private List<M3u8DownloadLine> m3u8DownloadLines;

	private int downloadIndex = 0;

	public String getM3u8Url() {
		return m3u8Url;
	}

	public void setM3u8Url(String m3u8Url) {
		this.m3u8Url = m3u8Url;
	}

	public List<M3u8DownloadLine> getM3u8DownloadLines() {
		return m3u8DownloadLines;
	}

	public void setM3u8DownloadLines(List<M3u8DownloadLine> m3u8DownloadLines) {
		this.m3u8DownloadLines = m3u8DownloadLines;
	}

	public void seekDownloadLine(String localPath) {
		synchronized (this) {
			for (int i = 0; i < m3u8DownloadLines.size(); i++) {
				M3u8DownloadLine downloadLine = m3u8DownloadLines.get(i);
				if (downloadLine.getLocalPath().equals(localPath)) {
					downloadIndex = i;
					break;
				}
			}
		}
	}

	public M3u8DownloadLine getNextDownloadLine() {
		synchronized (this) {
			if (downloadIndex < m3u8DownloadLines.size()) {
				M3u8DownloadLine next = m3u8DownloadLines.get(downloadIndex);
				downloadIndex++;
				return next;
			} else {
				return null;
			}
		}
	}

}
